pkg/tfbridge/property_path.go

```diff
@@ -125,3 +155,23 @@ func willTriggerReplacementRecursive(
 
 	return replacement
 }
+
+func schemaContainsComputed(
+	path propertyPath, rootTFSchema shim.SchemaMap, rootPulumiSchema map[string]*info.Schema,
+) bool {
+	computed := false
+	visitor := func(path walk.SchemaPath, tfs shim.Schema) {
+		if tfs.Computed() {
+			computed = true
+		}
+	}
+
+	tfs, _, err := lookupSchemas(path, rootTFSchema, rootPulumiSchema)
+	if err != nil {
+		return false
+	}
+
+	walk.VisitSchema(tfs, visitor)
+
```

Comment:
This is much easier to read if we break up the function into it's logical components:
1. Get the schema at `path`.
2. Check that schema for computed.

```suggestion
	tfs, _, err := lookupSchemas(path, rootTFSchema, rootPulumiSchema)
	if err != nil {
		return false
	}
	
	var computed bool
	walk.VisitSchema(tfs, func(_ walk.SchemaPath, tfs shim.Schema) {
		if tfs.Computed() {
			computed = true
		}
	})
```